libs/s25main/mapGenerator/Resources.cpp

```diff
@@ -177,39 +178,73 @@ namespace rttr { namespace mapGenerator {
     {
         const auto& textures = map.textureMap;
         auto& resources = map.resources;
+        struct MineableResourceInfo
+        {
+            // The current "budget", that is, how many many are we over, or under, compared to the desired distribution.
+            int budget = 0;
+            // The ratio is the user defined ratio for this resource.
+            int ratio;
+            // The resource encoding
+            libsiedler2::Resource resource;
+            MineableResourceInfo(int ratio, libsiedler2::Resource resource) : ratio(ratio), resource(resource) {}
+        };
+        MineableResourceInfo mRIs[4] = {MineableResourceInfo(settings.ratioCoal, libsiedler2::R_Coal),
+                                        MineableResourceInfo(settings.ratioGold, libsiedler2::R_Gold),
+                                        MineableResourceInfo(settings.ratioIron, libsiedler2::R_Iron),
+                                        MineableResourceInfo(settings.ratioGranite, libsiedler2::R_Granite)};
```

Comment:
Use a std::array here to avoid the magic number "4" in the resource picker function
```suggestion
        auto mRIs = helpers::make_array(MineableResourceInfo(settings.ratioCoal, libsiedler2::R_Coal),
                                                           MineableResourceInfo(settings.ratioGold, libsiedler2::R_Gold),
                                                           MineableResourceInfo(settings.ratioIron, libsiedler2::R_Iron),
                                                           MineableResourceInfo(settings.ratioGranite, libsiedler2::R_Granite));
```